README.md

```diff
@@ -0,0 +1,223 @@
+# Kernel Settings Role
+
+This role is used to modify kernel settings.  It uses `tuned` for its
+implementation, so it is able to manage all of the settings managed by `tuned`
+such as `sysctl`, `sysfs`, `vm`, `disk`, `net`, `modules`, and some of the
+`bootloader` settings such as `cmdline` and some `initrd` settings.  `tuned`
+is installed and enabled by default on Enterprise Linux and derivatives (RHEL
+and CentOS).  Using `tuned` allows managing all of the kernel parameters for
+performance tuning and security hardening, including bootloader parameters,
+all in one place (however it is no substitute for a general purpose
+bootloader/grub role).  `tuned` also persists settings upon reboot for those
+settings (such as `sysfs` settings) which would not otherwise be persisted (as
+opposed to `sysctl` which has an OS mechanism for applying settings at boot
+time).
+
+* `tuned` homepage - https://github.com/redhat-performance/tuned
+* `tuned` docs for EL7 - https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/performance_tuning_guide/chap-red_hat_enterprise_linux-performance_tuning_guide-tuned
+* `tuned` docs for EL8 - https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/monitoring_and_managing_system_status_and_performance/getting-started-with-tuned_monitoring-and-managing-system-status-and-performance
+
+## Requirements
+
+This role requires an operating system which has the `tuned` package and
+service.
+
+## Role Variables
+
+`kernel_settings_parameters` - default `{}` - This is a YAML dict of the
+settings you wish to apply, in `tuned` format.  For example:
+```yaml
+kernel_settings_parameters:
+  sysctl:
+    fs.epoll.max_user_watches: 785592
+    fs.file-max: 379724
+  sysfs:
+    /sys/kernel/debug/x86/pti_enabled: 0
+    /sys/kernel/debug/x86/retp_enabled: 0
+    /sys/kernel/debug/x86/ibrs_enabled: 0
+  vm:
+    transparent_hugepages: madvise
+  bootloader:
+    cmdline: spectre_v2=off nopti
+```
+See the `tuned` documentation above for the complete list of plugins `sysctl`,
+`sysfs`, etc., and the format of settings for those plugins.  *NOTE* that
+these settings are **additive**.  That is, they are applied **in addition to**
+any current settings.  For example, if you already had
+```yaml
+  sysctl:
+    kernel.threads-max: 29968
+    vm.max_map_count: 65530
+```
+then after applying the above, you would have
+```yaml
+  sysctl:
+    kernel.threads-max: 29968
+    vm.max_map_count: 65530
+    fs.epoll.max_user_watches: 785592
+    fs.file-max: 379724
+```
+This allows multiple higher level roles or playbooks to use this role to
+provide the kernel settings specific to that component.  For example, if you
+are installing a web server and a database server on the same machine, and
+they both require setting kernel parameters, the `kernel_settings_parameters`
```

Comment:
You're talking about the role itself here. I think `kernel_settings` was correct, wasn't it?

Reply:
yes